kf_lib_data_ingest/etl/ingest_pipeline.py

```diff
@@ -212,29 +217,44 @@ def check_stage_counts(self, stage):
         passed_all = True
 
         # Do stage counts validation
-        passed, message = stage_analyses.check_counts(
-            discovery_sources, self.data_ingest_config.expected_counts
+
+        # Split log messages to another file just for this step
```

Comment:
This doesn't really give us what we want. Right now the `counts_for_ingest.log` seems to be overwritten each time `check_stage_counts` is called and what we end up with is just the count logs for the GuidedTransformStage since that is the last stage for which count analysis actually runs. Additionally, it looks like only INFO level logs get output to this file.

Hmm I don't know if using a separate logger is the way to go here. I'm sorry if my comment was misleading when I gave the example with a `.log` ext. I think what we want is a count analysis test report which contains all of the detailed results of the count analysis for each stage (and perhaps even a summary at the top?). Not sure of the best way to accomplish that. As for the logs, I think what you're currently logging at INFO level is fine. And then maybe we just don't log anything additional at the DEBUG log level. Instead, that detailed info can be found in the count analysis test report.

My intention with this was just to avoid cluttering the ingest log with things that may be better in a separate report, and also make it easy for the user to see exactly what went wrong with count analysis by directing them to a report that *only* contains count analysis results.